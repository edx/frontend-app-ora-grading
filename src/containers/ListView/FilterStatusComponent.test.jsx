import React from 'react';
import { shallow } from 'enzyme';
import * as module from './FilterStatusComponent';

const mockFilterHooks = {
  clearFilters: jest.fn().mockName('clearFilters'),
  filterNames: ['some abitary', 'column'],
};

const mockSetAllFilters = jest.fn().mockName('setAllFilters');

describe('FilterStatusComponent component', () => {
  const props = {
    className: 'css-class-name',
    variant: 'button-variant',
    size: 'button-size',
    clearFiltersText: 'clear-filter-text',
    buttonClassName: 'css-class-name-for-button',
    showFilteredFields: true,
  };

  const contextProps = {
    state: {
      filters: [
        {
          id: 'filter-column',
          value: 'abitary',
        },
      ],
    },
    setAllFilters: () => mockSetAllFilters,
    headers: [
      {
        id: 'filter-column',
        Header: 'Formatted Filter Column',
      },
      {
        id: 'dummy-column',
        Header: 'Not showing column',
      },
    ],
  };
  const { FilterStatusComponent } = module;
  describe('snapshot', () => {
    test('with filter', () => {
      jest.spyOn(module, 'filterHooks').mockImplementationOnce(() => mockFilterHooks);
      const el = shallow(<FilterStatusComponent {...props} />);
      expect(el).toMatchSnapshot();
    });

    test('with filter but do not show filterd field', () => {
      jest.spyOn(module, 'filterHooks').mockImplementationOnce(() => mockFilterHooks);
      const el = shallow(<FilterStatusComponent {...props} showFilteredFields={false} />);
      expect(el).toMatchSnapshot();
    });

    test('without filter', () => {
      jest.spyOn(module, 'filterHooks').mockImplementationOnce(() => ({}));
      const el = shallow(<FilterStatusComponent {...props} />);
      expect(el).toMatchSnapshot();
    });
  });

  describe('component', () => {
    test('clear filter click', () => {
      jest.spyOn(module, 'filterHooks').mockImplementationOnce(() => mockFilterHooks);
      const el = shallow(<FilterStatusComponent {...props} />);
      el.find('Button').simulate('click');
      expect(mockFilterHooks.clearFilters).toHaveBeenCalledTimes(1);
    });

    test('on clear filter, set all filter get called with empty array', () => {
      jest.spyOn(React, 'useContext').mockImplementationOnce(() => ({ ...contextProps }));
      jest.spyOn(React, 'useCallback').mockImplementationOnce(cb => cb());
      const el = shallow(<FilterStatusComponent {...props} />);
      el.find('Button').simulate('click');
      expect(mockSetAllFilters).toHaveBeenCalledTimes(1);
    });
  });
});
