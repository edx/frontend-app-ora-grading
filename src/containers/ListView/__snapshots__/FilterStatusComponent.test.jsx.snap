// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterStatusComponent component snapshot with filter 1`] = `
<div
  className="css-class-name"
>
  <p>
    Filtered by 
    some abitary, column
  </p>
  <Button
    className="css-class-name-for-button"
    onClick={[MockFunction clearFilters]}
    size="button-size"
    variant="button-variant"
  >
    clear-filter-text
  </Button>
</div>
`;

exports[`FilterStatusComponent component snapshot with filter but do not show filterd field 1`] = `
<div
  className="css-class-name"
>
  <Button
    className="css-class-name-for-button"
    onClick={[MockFunction clearFilters]}
    size="button-size"
    variant="button-variant"
  >
    clear-filter-text
  </Button>
</div>
`;

exports[`FilterStatusComponent component snapshot without filter 1`] = `""`;
